ci/prow/e2e-incluster-build

```diff
@@ -5,38 +5,34 @@ set -euxo pipefail
 echo "Deploy KMMO..."
 make deploy
 
-echo "Wait until the KMMO Deployment is Available"
-timeout 1m kubectl wait --for condition=Available deployments.apps -n kmm-operator-system kmm-operator-controller-manager 
-
 echo "Create a build secret..."
 kubectl create secret generic build-secret --from-literal=ci-build-secret=super-secret-value
+
 echo "Add a configmap that contain the kernel module build dockerfile..."
 kubectl apply -f ci/kmm-kmod-dockerfile.yaml
 
 echo "Add an kmm-ci Module that contains a valid mapping..."
-sed -e "s/KVER_CHANGEME/$(uname -r)/g" ci/module-kmm-ci-build.template.yaml | tee module-kmm-ci.yaml
+sed -e "s/KVER_CHANGEME/$(minikube ssh -- uname -r)/g" ci/module-kmm-ci-build.template.yaml | tee module-kmm-ci.yaml
 kubectl apply -f module-kmm-ci.yaml
 
-echo "Wait for the job to be created..."
-timeout 1m bash -c 'until kubectl get job -l kmm.node.kubernetes.io/module.name | grep kmm; do sleep 1; done'
+# Wait for the build pod to be created. `kubectl wait` doesn't support such option,
+# see https://github.com/kubernetes/kubernetes/issues/83242.
+echo "Waiting for the build pod to be created..."
+timeout 1m bash -c 'until kubectl get pods -o json | jq -er ".items[].metadata.name | select(.? | match(\"build\"))"; do sleep 1; done'
+POD_NAME=$(kubectl get pods -o json | jq -r '.items[].metadata.name | select(.? | match("build"))')
 
-echo "Check that the module gets loaded on the node..."
-timeout 1m bash -c 'until minikube ssh -- lsmod | grep kmm_ci_a; do sleep 3; done'
+echo "Check that the build secret is mounted to the build pod"
+kubectl get pod/${POD_NAME} -o json | jq -e '.spec.containers[].volumeMounts[] | select(.name? | match("secret"))'
 
-echo "Check that the DriverContainer prints the secret's value to the standard output..."
-POD_NAME=$(kubectl get pod -l kmm.node.kubernetes.io/module.name --template='{{ (index .items 0).metadata.name }}')
-KERNEL_VERSION=$(uname -r)
-echo "Looking for the build secret"
-timeout 1m bash -c "until kubectl exec $POD_NAME -- grep super-secret-value /ci-build-secret; do sleep 3; done"
+# The build job/pod is deleted once done so we won't be able to get this info later on in the troubleshooting section.
+echo "Print the build logs..."
+# we can't get the logs on a pod that isn't `Running` yet.
+kubectl wait pod/${POD_NAME} --for jsonpath='{.status.phase}'=Running --timeout=60s
```

Comment:
isn't it a little bit problematic? We can fail it, since build pod may have been deleted before we check

Reply:
The build is pretty long because it is downloading the kernel-headers.